pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go

```diff
@@ -67,48 +70,82 @@ func (ba *BalancedAllocation) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewBalancedAllocation initializes a new plugin and returns it.
-func NewBalancedAllocation(_ runtime.Object, h framework.Handle) (framework.Plugin, error) {
+func NewBalancedAllocation(baArgs runtime.Object, h framework.Handle) (framework.Plugin, error) {
+	args, ok := baArgs.(*config.NodeResourcesBalancedAllocationArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourcesBalancedAllocationArgs, got %T", baArgs)
+	}
+
+	if err := validation.ValidateNodeResourcesBalancedAllocationArgs(nil, args); err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range (*args).Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+	}
+
 	return &BalancedAllocation{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
 			BalancedAllocationName,
 			balancedResourceScorer,
-			defaultRequestedRatioResources,
+			resToWeightMap,
 		},
 	}, nil
 }
 
-// todo: use resource weights in the scorer function
 func balancedResourceScorer(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
-	cpuFraction := fractionOfCapacity(requested[v1.ResourceCPU], allocable[v1.ResourceCPU])
-	memoryFraction := fractionOfCapacity(requested[v1.ResourceMemory], allocable[v1.ResourceMemory])
-	// This to find a node which has most balanced CPU, memory and volume usage.
-	if cpuFraction >= 1 || memoryFraction >= 1 {
-		// if requested >= capacity, the corresponding host should never be preferred.
-		return 0
+	resourceToFraction := make(map[v1.ResourceName]float64)
+	var totalFraction float64
+	for name, value := range requested {
+		if _, ok := allocable[name]; ok {
+			fraction := fractionOfCapacity(value, allocable[name])
+			if fraction >= 1 {
+				// if requested >= capacity, the corresponding host should never be preferred.
+				return 0
+			}
+			totalFraction = totalFraction + fraction
+			resourceToFraction[name] = fraction
+		} else {
+			klog.InfoS("requested resource not found for node score calculation", "resource", name)
```

Comment:
shouldn't we just not count it?

For example, if only some of my nodes have GPUs and I want them to be considered in the score if the nodes have  them, I have to configure the plugin to include GPUs. But the score should still give me something for the rest of the resources if the node doesn't have GPUs.

And since I expect this to be a common case, a log line seems like too noisy. It should be at least level 3.

Reply:
this is a little complicated, I need to think how this will impact the final score between the node with GPU and w/o GPU.

Reply:
Done